app/javascript/mastodon/service_worker/web_push_notifications.js

```diff
@@ -57,12 +57,16 @@ const openUrl = url =>
         .filter(client => /\/web\//.test(client.url))
         .sort(client => client !== 'visible');
 
-      const visibleClient = clientList.find(client => client.visibilityState === 'visible');
-      const focusedClient = clientList.find(client => client.focused);
+      if (webClients.length !== 0) {
+        return webClients[0].focus().then(client => client.postMessage({ type: 'navigate', url }));
+      } else {
+        const visibleClient = clientList.find(client => client.visibilityState === 'visible');
+        const focusedClient = clientList.find(client => client.focused);
```

Comment:
If you add the website to the home-screen and go to the settings, the notification should open the "app", not Chrome with a new tab.